docs/current_docs/guides/578291-build-test-publish-java-spring.mdx

```diff
@@ -0,0 +1,124 @@
+---
+slug: /guides/578291/build-test-publish-java-spring
+displayed_sidebar: "current"
+category: "guides"
+tags: ["java", "spring"]
+authors: ["Vikram Vaswani"]
+date: "2024-04-04"
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Build, Test and Publish a Spring Application with Dagger
+
+## Introduction
+
+This tutorial explains how to use Dagger to build, test and publish a Java application using Spring. You will learn how to create Dagger Functions to:
+- Build your Spring application with all required dependencies
+- Run unit tests for your Spring application
+- Publish the final application image to Docker Hub
+
+## Step 1: Initialize a new module
+
+The example module used in this guide builds, tests and publishes a Spring application.
+
+Run `dagger init` in the application directory to bootstrap a new module:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```sh
+dagger init --name=my-module --sdk=go
+```
+
+This will generate a `dagger.json` module file, an initial `dagger/main.go` source file, as well as a `dagger/dagger.gen.go` file and `dagger/internal/` directory.
+
+</TabItem>
+<TabItem value="Python">
+
+```sh
+dagger init --name=my-module --sdk=python
+```
+
+This will generate a `dagger.json` module file, initial `dagger/src/main.py` and `dagger/pyproject.toml` files, as well as a generated `dagger/sdk` folder for local development.
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```sh
+dagger init --name=my-module --sdk=typescript
+```
+
+This will generate a `dagger.json` module file, initial `dagger/src/index.ts`, `dagger/package.json` and `dagger/tsconfig.json` files, as well as a generated `dagger/sdk` folder for local development.
+
+</TabItem>
+</Tabs>
+
+## Step 2: Add dependencies
+
+```shell
+dagger install github.com/levlaz/daggerverse/mariadb@v0.2.1
+dagger install github.com/jcsirot/daggerverse/java@250eb3e0f49da321cca2241fe1e1d017b4143a91
+dagger install github.com/shykes/daggerverse/docker@3338120927f8e291c4780de691ef63a7c9d825c0
```

Comment:
What do we need Docker for? That seems really out of left field, I'd only expect to run Docker in Dagger in fringe situations (legacy compatibility, booth demos, etc).

Removed it locally, doesn't seem to affect anything :thinking:

Reply:
@vikram-dagger so I got it past the broken DB setup, but now it's failing in a few places because the `docker` CLI isn't installed. Pausing there for now. Left inline comments covering my tracks.

To be honest if this example requires Docker it feels pretty distracting from the core focus of the doc (see my initial reaction). I wonder if we should just find a simpler example.

Reply:
@vito thanks for your help on this, you made much more progress than I did! I'm fine with using a different Spring or other Java example. Do you have a suggestion, or should I try to find something online?